main/src/test/java/org/mobilitydata/gtfsvalidator/validator/GtfsFrequencyEndRangeValidatorTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.StartAndEndRangeEqualNotice;
+import org.mobilitydata.gtfsvalidator.notice.StartAndEndRangeOutOfOrderNotice;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsFrequency;
+import org.mobilitydata.gtfsvalidator.table.GtfsFrequencyTableLoader;
+import org.mobilitydata.gtfsvalidator.type.GtfsTime;
+
+// FIXME: Cover autogenerated end-range validators by unit tests in "processor".
+
+@RunWith(JUnit4.class)
+public class GtfsFrequencyEndRangeValidatorTest {
+
+  private static List<ValidationNotice> generateNotices(GtfsFrequency frequency) {
+    NoticeContainer noticeContainer = new NoticeContainer();
+    GtfsFrequencyEndRangeValidator validator = new GtfsFrequencyEndRangeValidator();
+    validator.validate(frequency, noticeContainer);
+    return noticeContainer.getValidationNotices();
+  }
+
+  @Test
+  public void inOrder_yieldsNoNotice() {
+    assertThat(
+            generateNotices(
+                new GtfsFrequency.Builder()
+                    .setCsvRowNumber(2)
+                    .setStartTime(GtfsTime.fromString("10:00:00"))
+                    .setEndTime(GtfsTime.fromString("11:00:00"))
+                    .build()))
+        .isEmpty();
+  }
+
+  @Test
+  public void outOfOrder_yieldsNotice() {
```

Comment:
```suggestion
  public void timesOutOfOrderShouldYieldsNotice() {
```

Reply:
I am following the Google style guide here. Words like "should" are redundant. Underscores make test names more readable: they separate the actions taken from the expected outcome.

The Google style guide says:

A test's name should summarize the behavior it is testing. A good name describes both the actions that are being taken on a system and the expected outcome. Test names will sometimes include additional information like the state of the system or its environment before taking action on it.

* transferMoney_deductsFromSource
* transferMoney_insufficientFunds_throwsException
* multiply_twoNegativeValues_returnsPositiveValue
* isInputValid_conversionSucceeds

Reply:
Makes sense to me! @barbeau?

Reply:
I agree that the underscore pattern is easier to read, and is allowed by the Google Java Style Guide that we've previously agreed upon:
https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names

>Underscores may appear in JUnit test method names to separate logical components of the name, with each component written in lowerCamelCase. One typical pattern is <methodUnderTest>_<state>, for example pop_emptyStack. There is no One Correct Way to name test methods.

I would propose we follow this pattern going forward. We can refactor existing tests as we touch them.